### PR TITLE
(Android) Play Store builds: Do not resolve symlinks when handling core paths

### PR DESCRIPTION
**Pull request by jdgleaver**
## Description

When handling core file paths. RetroArch typically resolves symbolic links - this is done (mainly) to prevent duplicate playlist entries when running on a symlinked filesystem (i.e. when using symbolic links, it is very easy to end up in situations where the same core is referenced by multiple different path text strings, which confuses the detection of identical entries in the playlist code).

This, however, fails completely when using the Android Play Store builds. In this case, the core shared object files installed via play feature delivery are placed in arbitrary filesystem locations and then symlinked into RetroArch's `cores` folder - but these feature delivery files have non-standard names, with the format `lib<core_name>.so`.

On Android, RetroArch expects core files with the name `<core_name>_libretro_android.so` - and the core symlinks respect this. But as soon as we resolve one of these symlinks, we end up with the 'invalid' source file name, which breaks several aspects of the core file handling code.

This PR very simply modifies the code such that core file symlinks *are not* resolved when using the Android Play Store builds, so core names are detected/handled correctly.

In addition, since we are now calling `play_feature_delivery_enabled()` (to identify Play Store builds) with greater frequency, the function has been optimised - instead of calling a (very slow) Java method, we now cache the result on first use, and return the cached value (with appropriate mutex locking).

### CODE REVIEW

**Comment by i30817** [comment]
Why is it 'very easy' to end up with both symlinked and normal versions of the same file in other platforms? Shouldn't be this consistent and only use the symlink everywhere?


It's not like retroarch has multiple directories to search for cores...

**Comment by farmerbb** [comment]
Tested this branch on my private Play Store listing, works great!  No issues that I can see.

**Comment by jdgleaver** [comment]
@farmerbb Thanks! Much appreciated, as always :)

@i30817 You might have a point there...

RetroArch has always used real paths when loading dynamic libraries, I am unsure of the historical reasons for this (perhaps some oddball platforms have issues with symlinked libraries?), but then I do remember one dev insisting that we have a method for uniquely identifying multiple copies of the same core (for debugging purposes) - so whatever happened, I guess 'real paths' stuck.

There was a time when playlists recorded whatever core path they were given, and since RetroArch was converting to real paths between content load and playlist update, this guaranteed duplicate history playlist entries when the same content was launched using different methods (playlist vs load content) on a symlinked filesystem. This happened *all the time*, and was very annoying. When I fixed this, again there was contention over identifying different copies of the same core, so I just ended up changing everything to real paths. This works fine, and would never have been an issue if we didn't have to support play store builds :P

Now the thing is, we've since modified playlist content loading such that it always searches the core info struct to 'sanitise' the core path string - so real paths in playlists have become a bit of a moot point. I guess we could therefore scrap real paths entirely - but then that's a lot of well-tested code to modify, with potential for regressions, and there's also a concern over playlist backwards compatibility (i.e. oldish versions of RetroArch expect playlists with real paths, so loading a newly generated playlist containing symlink paths with an old version of RA will bring back the duplicate entries issue).

In short, it probably makes sense to rethink real core paths. but it needs some more consideration. I'll add it to the TODO list :)
